src/main/java/io/supertokens/webserver/api/session/SessionRemoveAPI.java

```diff
@@ -113,10 +113,18 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
                 String[] sessionHandlesRevoked;
 
                 if (revokeAcrossAllTenants) {
+                    // when revokeAcrossAllTenants is true, and given that the backend SDK might pass tenant id
+                    // we do not want to enfore public tenant here but behave as if this is an app specific API
                     AppIdentifier appIdentifier = getAppIdentifier(req);
-                    StorageAndUserIdMapping storageAndUserIdMapping = getStorageAndUserIdMappingForAppSpecificApiWithoutEnforcingPublicTenant(
-                            req, userId, UserIdType.ANY);
-
+                    Storage[] storages = StorageLayer.getStoragesForApp(main, appIdentifier);
```

Comment:
add comment explaining why we are doing this here and not calling any of the functions defined in the super class

Reply:
Done